### ISSUE TITLE
[BUG] vSphere node driver cluster fails to provision when the nodes are cloned from a VM

### ISSUE THREAD

**Issue by rishabhmsra** [rancher/rancher#38998]
**Rancher Server Setup**
- Rancher version: v2.6-head(a3b5237)
- Installation option (Docker install/Helm Chart): Docker

**Information about the Cluster**
- Kubernetes version: v1.24.4+rke2r1

- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): vSphere node driver(1-cp, 1-etcd, 1-w)

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin
  - If custom, define the set of permissions:

**Describe the bug**

- In vSphere 7 environment, created a node driver cluster(1-cp, 1-etcd, 1-w) and cloned the nodes from an existing VM. Cluster fails to provision with error: `Configuring control plane node(s) rke2-clone-rishabhcp-clone-84755c54f7-m2pks: waiting for probes: calico, kube-apiserver, kube-controller-manager, kube-scheduler, kubelet`
- On vSphere cp, etcd and worker node got created but same IP got assigned to all the 3 nodes.
- Also not able to SSH into the nodes, it errors out with : `client_loop: send disconnect: Broken pipe`
- See the steps below

**To Reproduce**

- On vSphere 7 env, create a rancher server
- Create a VM on vSphere
- On rancher server provision a node driver cluster (1-cp, 1-etcd, 1-w) by selecting `Clone an existing virtual machine` -> `Virtual Machine` -> `select the VM you created in the drop down`
![instance-option](https://user-images.githubusercontent.com/36376154/191026813-1938d841-86bd-4f9a-b505-06f98f034ab5.png)
- Observe that in vSphere cp, etcd, and worker nodes will be created but same IP gets assigned to all 3 nodes.

**Result**
- Cluster fails to Provision when nodes are cloned from a VM.

**Expected Result**

- Cluster should come into Active state.

**Comment by slickwarren** [comment]
I don’t see much about it in our docs, but I did [find this article which is helpful](https://support.us.ovhcloud.com/hc/en-us/articles/360017132080-How-to-Clone-a-VM-in-vSphere).
tldr of that article
```
Warning: If you did not customize the VM, you will need to change the clone configuration before you boot it up to avoid an IP/MAC conflict. To do so, uncheck the network adapter in the VM settings once it has been cloned.
```

Our testing setup does not have a multi-network setup in order to properly clone the VM without having IP/MAC conflicts, hence why you are getting no IPs in this use case.  I would typically suggest something adding some warning to the rancher UI, but that would be doing more than what Vsphere’s UI does when cloning, so I doubt that would make much sense to do. 

Closing this ticket due to misconfiguration / user error.
